src/components/bottomSheet/bottom-sheet.js

```diff
@@ -175,24 +176,37 @@ function MdBottomSheetProvider($$interimElementProvider) {
         options.restoreScroll = $mdUtil.disableScrollAround(bottomSheet.element, options.parent);
       }
 
-      return $animate.enter(bottomSheet.element, options.parent)
-        .then(function() {
-          var focusable = $mdUtil.findFocusTarget(element) || angular.element(
-            element[0].querySelector('button') ||
-            element[0].querySelector('a') ||
-            element[0].querySelector('[ng-click]')
-          );
-          focusable.focus();
-
-          if (options.escapeToClose) {
-            options.rootElementKeyupCallback = function(e) {
-              if (e.keyCode === $mdConstant.KEY_CODE.ESCAPE) {
-                $mdUtil.nextTick($mdBottomSheet.cancel,true);
-              }
-            };
-            $rootElement.on('keyup', options.rootElementKeyupCallback);
-          }
-        });
+      element.addClass('md-prepare-enter');
+      options.parent.append(element);
+
+      // Wait for element rendering below bottom
+      $timeout(function() {
+
+        // manually trigger the animation
+        $timeout(function() {
+          element.removeClass('md-prepare-enter');
+        }, SWIFT_EASE_OUT_DURATION);
+
+        $animate.enter(bottomSheet.element, options.parent)
+          .then(function() {
+
+            var focusable = $mdUtil.findFocusTarget(element) || angular.element(
+                element[0].querySelector('button') ||
+                element[0].querySelector('a') ||
+                element[0].querySelector('[ng-click]')
+              );
+            focusable.focus();
+
+            if (options.escapeToClose) {
+              options.rootElementKeyupCallback = function(e) {
+                if (e.keyCode === $mdConstant.KEY_CODE.ESCAPE) {
+                  $mdUtil.nextTick($mdBottomSheet.cancel,true);
+                }
+              };
+              $rootElement.on('keyup', options.rootElementKeyupCallback);
+            }
+          });
+      }, 10);
```

Comment:
Yes we should give the element a free time to complete the insertion, so we can run the animation clean.